tests/test_authz_decorator.py

```diff
@@ -100,3 +104,47 @@ def handler(self, restrictions):
         auth_header = Authorizer.sign_authz({"allow": {"x": "*"}, "deny": {}}, SAMPLE_PRIVATE_KEY)
         resp = XResource({**event, "headers": {"authorization": auth_header}})()
         assert resp.status_code == HTTPStatus.FORBIDDEN
+
+    def test_check_permission(self) -> None:
```

Comment:
You are in the `TestAuthorizationDecorator` class, you shouldn't put here tests for the `check_permission` function. Especially that they look like duplicates of the tests which I see above 🤔

Reply:
Also, I don't see the test which proves that the bug is fixed. I think the simplest solution here is to follow the currently chosen pattern and test the fix via the `authorization` decorator, and improve in the future.

Reply:
Something like:
```python
@patch.dict(environ, env_mock)
def test_authorization_header_takes_priority_over_guest_policy_when_both_present(self, *_args):
    class XResource(Resource):
        @add_route("/")
        @authorization()
        def handler(self, restrictions):
            return Response("x")

        @staticmethod
        def get_guest_authorization() -> dict:
            return {"allow": {}, "deny": {}}

    auth_header = Authorizer.sign_authz({"allow": "*", "deny": {}}, SAMPLE_PRIVATE_KEY)
    resp = XResource({**event, "headers": {"authorization": auth_header}})()
    assert resp.status_code == 200
```

Runt this test with both the old and the new code, you will see that it is passing only after the fix 😉